src/environment/ale_state.cpp

```diff
@@ -259,6 +259,19 @@ void ALEState::applyActionPaddles(Event* event, int player_a_action, int player_
   }
 }
 
+void ALEState::pressSelect(Event* event) {
+  resetKeys(event);
+  event->set(Event::ConsoleSelect, 1);
+}
+
+void ALEState::setDifficulty(Event* event, unsigned mask) {
```

Comment:
unsigned should be unsigned int. Is there a reason to use a mask, rather than two booleans or integers (one per player)? With a mask it won't be obvious what the function arguments do.

Reply:
The mask makes some of the code more transparent. This is because the difficulty switch was a 0 and 1, and when we request getDifficulties it returns that.